testing/e2e-testing.md

```diff
@@ -0,0 +1,100 @@
+3) Exporter error. The exporter could fail to  account for resources properly.
+4) Scheduler error. The scheduler could choose a node that cannot meet topology constraints.
+5) Kubelet misreporting. The kubelet could report the wrong resources at its end point.
+6) Kubelet unavailable. The kubelet's information may not be available at the PodResources endpoint.
+7) Scheduler unavailable. The Topology Scheduler may be unavailable to the cluster.
+8) Exporter error. The exporter could crash leading to no updates for one or more nodes.
+
+
+### Test cases
+
+| # | Test target 	| Component | Failure state	 		| 		 	|
+|---|-------------------|-----------|-----------------------------------|-------------------------------|
+| 1 | |  |	  	| 		|
+| 2 | |  |  		| 		|
+| 3 | |  | 		|  		|
+| 4 | |  | 		| 		|
+| 5 | |  |	   	|  		|
+
+## Performance testing
+
+### Points of failure
+Points of failure for performance testing are different from those in a purely functional test. Initially the performance tests will look for two kinds of failures:
+1) Latency should be recorded for multiple different scales and levels of activity. Latency for the differenct components - scheduler, api server etc. may be a limiting factor on scaling of Topology Aware Scheduling.
+2) Overload, where the system starts to show errors when utilization is high, should also be tested at scale. The major types of error, the distribution of errors and the scale at which those errors occur will be the backbone of overload testing.
+
+### Test cases
+### Frameworks for performance testing at scale
+In order to test cluster sizes of up to 5000 nodes simulation of some cluster nodes are needed. Testing is to be carried out with a High Availability control plane of 3 bare metal nodes. In future a set up with 3 Control Plane nodes and an external HA etcd cluster will be considered. Cloud providers do not in general offer instances with multiple NUMA zones, meaning that this type of test will have to be carried out in an on-prem environment.
+
+Scales used for testing:
+**Toy:** 10 Nodes
+**Small:** 100 Nodes
+**Medium:** 500 Nodes
+**Large:** 1000 Nodes
+**Huge:** 5000 Nodes
+
+The following options are under consideration for end to end testing at scale:
+
+#### 1) VMs
+It is relatively easy to create a virtual machine that acts as a "real" Kubernetes node and this is common practice in production Kubernetes. It is also possible to create virtual machines that use resources across NUMA zones i.e. each VM with 4 cpus, two from each NUMA zone. The tradeoff here is between number of nodes and amount of schedulable resources on a node. 
+
+Additional orchestration will also be needed in order to bring up a large number of virtual machines in a bare-metal environment. This additional overhead is likely to make end to end testing using virtual machines the most labor-intensive option, and the one that is least likely to be easily shareable across the community. 
+
+#### 2) Kubernetes in Docker
```

Comment:
Frankly saying, I don't know real working solution for kubelet inside docker container.